pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -257,6 +221,54 @@ func (az *Cloud) EnsureLoadBalancer(clusterName string, service *v1.Service, nod
 		}
 	}
 
+	var lbStatus *v1.LoadBalancerStatus
+	if lbIP == nil {
+		lbStatus, exists, err := az.GetLoadBalancer(clusterName, service)
+		if err != nil {
+			return nil, err
+		}
+		if !exists {
+			return nil, fmt.Errorf("ensure(%s): lb(%s) - failed to get back load balancer", serviceName, lbName)
+		}
+		lbIP = &lbStatus.Ingress[0].IP
+	}
+
+	az.operationPollRateLimiter.Accept()
+	glog.V(10).Infof("SecurityGroupsClient.Get(%q): start", az.SecurityGroupName)
+	sg, err := az.SecurityGroupsClient.Get(az.ResourceGroup, az.SecurityGroupName, "")
+	glog.V(10).Infof("SecurityGroupsClient.Get(%q): end", az.SecurityGroupName)
+	if err != nil {
+		return nil, err
+	}
+	sg, sgNeedsUpdate, err := az.reconcileSecurityGroup(sg, clusterName, service, lbIP, true /* wantLb */)
```

Comment:
Does locking down to the lbIP affect the delivery of the external IP to the node?  The LB is set to "enableFloatingIP": true, and I believe this drops the external IP address on the box.

Reply:
I've tested the following scenarios:

* Single node so the request is guaranteed to land on the right node
* Multiple nodes so kubeproxy has to forward the request to the right node
* Initiating a request from within the cluster to exercise the Node->LB->Node route

All of these are working, though I've only tested interactively.

Reply:
From our conversation I think this addresses the routing cases that you wanted to check - let me know if I'm still not covering the bases!

Reply:
Can you also check to ensure that the source IP preservation still works with this:

https://kubernetes.io/docs/tutorials/services/source-ip/

Thanks!

Reply:
I ran a test using the `source-ip-app` tool from the page you linked.  Without the `externalTrafficPolicy: Local` setting, the tool reported `client_address=10.244.0.1`.  With the `externalTrafficPolicy: Local` setting, the tool reported `client_address=167.220.242.83`.  So I think source IP preservation is still okay.  Thanks for flagging it!

(I'm still not sure how to automate this testing though.  It feels like our testing matrix is getting more complex by the day and ad hoc hand testing won't cut it forever...)